src/lib.rs

```diff
@@ -557,25 +608,22 @@ impl<'a> AggregatedQueries<'a> {
                     .map(|rule_listener_query| rule_listener_query.resolve(language.language()))
                     .enumerate()
                 {
+                    let capture_index_if_per_capture: CaptureIndexIfPerCapture =
```

Comment:
So sort of similarly to how the listener queries are "registered" separately from (the single method) where they get invoked, we keep track "up front" of per-listener whether they are "match-oriented" vs "single-capture-oriented" and then basically around that point of calling into that single "there was a listener match" callback we construct an enum value that will either be a "single capture" vs an "entire match" based on what they "requested"/"registered"

And then similar to how it is "under the hood" (ie beyond that single method entry point) that implementors presumably "de-multiplex" (?)/"route" that call to an individual listener callback, they may "unwrap" that enum value based on whether it was registered as single-capture-oriented vs match-oriented

A little "particular" but especially if usually wired up for you via a macro then I think it probably makes rough sense as a "boundary"